### PR TITLE
Generate settings manifest

### PR DESCRIPTION
**Pull request by bion**
### Description

Adds python code for generating the `ServerSettings.java` file described in [admin settings panel design](https://docs.google.com/document/d/1FK5YLJd6jVu1O1uuu2OOR2mXTs3yVEXkzyfjVwnkz8s/edit).

Does not consume the generated code. The next PR will though.

The logic for retrieving config values won't work until https://github.com/civiform/civiform/pull/4982 is merged.

#### General

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [x] Created tests which fail without the change (if possible)

### Issue(s) this completes

Contributes to https://github.com/civiform/civiform/issues/4948

### CODE REVIEW

File: server/app/services/settings/SettingsManifest.java
Comment (rockycodes) [inline server/app/services/settings/SettingsManifest.java:48]: nitpick (non-blocking): regenerate before merging to remove this var that was removed in your other pr.

**Review by rockycodes — APPROVED** [review]
This looks good to me! It's such a cool setup. I am not the most well versed in Java and Python, so recommend getting another set of eyes if you would like more feedback/suggestions on your code. But overall the logic/setup is clear and makes sense to me, just a couple of nitpicks.

File: server/app/services/settings/AbstractSettingsManifest.java
Comment (reemaka) [inline server/app/services/settings/AbstractSettingsManifest.java:53]: What's "Hocon"?

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py
Comment (bion) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:13]: Done

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py
Comment (bion) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:1]: Done

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py
Comment (bion) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:55]: Done

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py
Comment (bion) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:57]: Done

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py
Comment (bion) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:79]: Done

File: server/app/services/settings/SettingsManifest.java
Comment (bion) [reply server/app/services/settings/SettingsManifest.java:48]: Done

File: env-var-docs/parser-package/tests/settings_manifest_test.py
Comment (bion) [reply env-var-docs/parser-package/tests/settings_manifest_test.py]: Done

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py
Comment (reemaka) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:15]: Lol the link was supposed to be: https://google.github.io/styleguide/pyguide.html#313-imports-formatting, oops

**Review by reemaka — APPROVED** [review]

File: env-var-docs/parser-package/src/env_var_docs/settings_manifest.py
Comment (bion) [reply env-var-docs/parser-package/src/env_var_docs/settings_manifest.py:15]: I was wondering 😆

**Pull request merged**
